ui-ngx/src/app/modules/home/components/widget/lib/gateway/connectors-configuration/modbus/modbus-security-config/modbus-security-config.component.ts

```diff
@@ -0,0 +1,147 @@
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+
+import { ChangeDetectionStrategy, Component, forwardRef, Input, OnChanges, OnDestroy } from '@angular/core';
+import {
+  ControlValueAccessor,
+  FormBuilder,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  UntypedFormGroup,
+  ValidationErrors,
+  Validator,
+  Validators
+} from '@angular/forms';
+import {
+  ModbusSecurity,
+  noLeadTrailSpacesRegex,
+} from '@home/components/widget/lib/gateway/gateway-widget.models';
+import { SharedModule } from '@shared/shared.module';
+import { CommonModule } from '@angular/common';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import {
+  SecurityConfigComponent
+} from '@home/components/widget/lib/gateway/connectors-configuration/security-config/security-config.component';
+
+@Component({
+  selector: 'tb-modbus-security-config',
+  templateUrl: './modbus-security-config.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => ModbusSecurityConfigComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => ModbusSecurityConfigComponent),
+      multi: true
+    }
+  ],
+  standalone: true,
+  imports: [
+    CommonModule,
+    SharedModule,
+    SecurityConfigComponent,
+  ]
+})
+export class ModbusSecurityConfigComponent implements ControlValueAccessor, Validator, OnChanges, OnDestroy {
+
+  @Input() isMaster = false;
+  @Input() disabled = false;
+
+  securityConfigFormGroup: UntypedFormGroup;
+
+  private onChange: (value: ModbusSecurity) => void;
+  private onTouched: () => void;
+
+  private destroy$ = new Subject<void>();
+
+  constructor(private fb: FormBuilder) {
+    this.securityConfigFormGroup = this.fb.group({
+      certfile: ['', [Validators.pattern(noLeadTrailSpacesRegex)]],
+      keyfile: ['', [Validators.pattern(noLeadTrailSpacesRegex)]],
+      password: ['', [Validators.pattern(noLeadTrailSpacesRegex)]],
+      server_hostname: ['', [Validators.pattern(noLeadTrailSpacesRegex)]],
+    });
+
+    this.observeValueChanges();
+  }
+
+  ngOnChanges(): void {
+    if (this.isMaster) {
+      this.securityConfigFormGroup = this.fb.group({
+        certfile: ['', [Validators.pattern(noLeadTrailSpacesRegex)]],
+        keyfile: ['', [Validators.pattern(noLeadTrailSpacesRegex)]],
+        password: ['', [Validators.pattern(noLeadTrailSpacesRegex)]],
+        reqclicert: [false, []],
+      });
+      this.observeValueChanges();
+    }
+    if (this.disabled) {
+      this.securityConfigFormGroup.disable({emitEvent:false});
+    } else {
+      this.securityConfigFormGroup.enable({emitEvent:false});
+    }
```

Comment:
This is an antipattern. You need to implement the method setDisabledState with ControlValueAccessor.